arch/risc-v/src/common/supervisor/riscv_syscall_dispatch.S

```diff
@@ -0,0 +1,148 @@
+/****************************************************************************
+ * arch/risc-v/src/common/supervisor/riscv_syscall_dispatch.S
```

Comment:
change to riscv_syscall_dispatch to riscv_dispatch_syscall like riscv_dispatch_irq

Reply:
Fix by #5959.